File: Store.BusinessLogic/Models/Filters/PrintingEditionFilterModel.cs
using System.Collections.Generic;
using Store.BusinessLogic.Models.Base;
using Store.DataAccess.Entities.Enums;

namespace Store.BusinessLogic.Models.Filters
{
    public class PrintingEditionFilterModel : BaseFilterModel
    {
        public string SearchQuery { get; set; }
        public decimal? MinPrice { get; set; }
        public decimal? MaxPrice { get; set; }
        public IList<int> Types { get; set; }
        public Enums.PrintingEditions.Currency Currency { get; set; } = Enums.PrintingEditions.Currency.USD;
    }
}

File: Store.View/src/app/shared/models/printing-edition/printing-edition.filter.model.ts
import { BaseFilterModel } from 'src/app/shared/models/base/base-filter.model';
import { PrintingEditionType, PrintingEditionCurrency } from 'src/app/shared/enums';

export class PrintingEditionFilterModel extends BaseFilterModel {
    public searchQuery?: string;
    public types?: Array<PrintingEditionType>;
    public minPrice?: number;
    public maxPrice?: number;
    public currency?: PrintingEditionCurrency;
}
